File: ListaEnlazada.java
public class ListaEnlazada {
private Nodo primero;
private int numeroElementos;

public static void main(String[] args) {
ListaEnlazada obj1 = new ListaEnlazada();
obj1.estaVacia();
obj1.insertarElemento(7);
obj1.insertarElemento(4);
obj1.insertarElemento(0);
obj1.insertarElemento(10);
obj1.insertarElemento(5);
System.out.println("");
obj1.listar();
System.out.println("");
obj1.estaVacia();
}

public ListaEnlazada() {
this.primero = null;
this.numeroElementos = 0;
}

// validar si la lista esta vacia
public void estaVacia() {
if(primero == null)
   System.out.println("La lista esta vacia.");
else
   System.out.println("La lista tiene " + numeroElementos + " elementos.");
}

// agregar elementos a la lista
public void insertarElemento(int dato) {
Nodo nuevo = new Nodo(dato);
nuevo.siguiente = primero;
primero = nuevo;
numeroElementos++;
}

// listamos la lista
public void listar() {
Nodo actual = primero;
while(actual != null) {
System.out.print("[" + actual.dato + "]--->");
actual = actual.siguiente;
}
System.out.println("null");
}

}

File: ListaEnlazada.cpp
#include <iostream>
#include <stdlib.h>

class Nodo {
	public:
		Nodo *siguiente;
		int dato;
		Nodo(int dato){
			this->dato = dato;
			this->siguiente = NULL;
		}
};

class ListaEnlazada{
	private:
		Nodo *primero;
		int numeroElementos;
	public:
		ListaEnlazada(){
			this->primero = NULL;
			this->numeroElementos = 0;
		}
		
		void estaVacia() {
			if(primero == NULL)
				std::cout << "La lista esta vacia." << std::endl;
			else
				std::cout << "La lista tiene " << numeroElementos << " elementos." << std::endl;
		}
		
		void insertarDatos(int dato) {
			Nodo *nuevo = new Nodo(dato);
			nuevo->siguiente = primero;
			primero = nuevo;
			numeroElementos++;
		}
		
		void listar() {
			Nodo *actual = primero;
			while(actual != NULL) {
				std::cout << "[" << actual->dato << "]--->";
				actual = actual->siguiente;
			}
			std::cout << "NULL" << std::endl;
		}
};

int main(void) {
	ListaEnlazada *obj1 = new ListaEnlazada();
	obj1->estaVacia();
	std::cout << "" << std::endl;
	obj1->insertarDatos(8);
	obj1->insertarDatos(15);
	obj1->insertarDatos(36);
	obj1->insertarDatos(23);
	obj1->insertarDatos(1);
	obj1->listar();
	std::cout << "" << std::endl;
	obj1->estaVacia();
	std::cout << "" << std::endl;
	
	
	system("pause > 0");
	return EXIT_SUCCESS;
}
